guarddog/scanners/go_project_scanner.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import requests
+
+from guarddog.scanners.go_package_scanner import (
+    GoModuleScanner,
+    GOPROXY_URL,
+    escape_module_name,
+)
+from guarddog.scanners.scanner import ProjectScanner
+
+log = logging.getLogger("guarddog")
+
+
+class GoModule:
+    def __init__(self):
+        self.module = ""
+        self.go = ""
+        self.toolchain = ""
+        self.requirements = []
+
+
+class Requirement:
+    def __init__(self, module: str, version: str):
+        self.module = module
+        self.version = version
+
+
+class GoDependenciesScanner(ProjectScanner):
+    def __init__(self) -> None:
+        super().__init__(GoModuleScanner())
+
+    def parse_requirements(self, raw_requirements: str) -> dict[str, set[str]]:
+        main_mod = self.parse_go_mod_file(raw_requirements)
+        # TODO: implement the Minimal Version Selection algorithm as described
+        # In https://research.swtch.com/vgo-mvs and https://go.dev/ref/mod#minimal-version-selection
```

Comment:
correct me if i'm wrong but Guarddog don't do resolve dependencies on dependencies (transitive).
So, MVS would not play here.

our go.mod might look like this:
```
require packageA version1
require packageB version2
```

if both packages then depend on packageC, then MVS will resolve which is the minimum common version to use.
Since Guarddog don't resolve indirect dependencies, this is not required

Reply:
Ok thanks for confirming! I was not sure what exactly the `verify` command meant to be fair. Is it planned for guarddog to resolve transitive dependencies someday though?

Reply:
is definitely something to consider, since transitive deps are a bit part of supply chain attacks.
I'll discuss this feature internally.